### ISSUE TITLE
v5.3.1RC2 results in duplicate devices after adding a new device

### ISSUE THREAD

**Issue by K1WIZ** [bwssytems/ha-bridge#1258]
Just recently upgraded to 5.3.1 RC2  to resolve the inability in discovering devices.

New problem:

I had 22 devices on my installation all added fine initially after upgrading.  I added a 23rd and now I have duplicates for every device - a total of 46 in all.   It seems there may be another bug in this.   I have had to go into alexa web interface on Amazon and remove all devices and re-add them to get the 23.   

The older version (when it worked) remembered what devices already existed and only added the one new device.

**Comment by gohamstergo** [comment]
Similar here. I removed all devices from Alexa and when I ran discovery, it discovered everything twice.

**Comment by bwssytems** [comment]
Was there another echo active at the time the devices were removed?

**Comment by K1WIZ** [comment]
I have Alexa speakers in every room of the house.   (a total of 7).  I never had any problems adding new devices until now on this release.  

previously on older releases, I would simply create the new device in Domoticz, then add it to habridge, then do discovery and Alexa would recognize one new added device and allow me to add it to the already existing devices.

on 5.3.1 RC2 the behavior is different:
add new device to domoticz
add new domoticz device to habridge
run discovery on alexa
end up with doubles of what was there before plus the new device.

I got around it by doing:
delete all habridge devices on alexa.amazon.com web portal
run discovery again
correct number of devices now detected (and have to re-add to groups etc)

It seems the only way to add a single or more new devices is to do:
add new device to domoticz
add new domoticz device to habridge
delete all habridge hosted devices from alexa.amazon.com portal
rediscover devices
re-add all devices to appropriate groups.  

In short, it seems like there is no duplicate detection in the process.

**Comment by ypoosn** [comment]
> Similar here. I removed all devices from Alexa and when I ran discovery, it discovered everything twice.
> 
> edit: i removed all devices a second time, unplugged one of my Echos (leaving only one active) and everything was discovered just once. I do have the link button disabled since my echos dont ever get picked up by that anymore, so that could perhaps be part of it.

The same for me after unplugging all echos except one. Removing all devices and rediscovering with the remaining echo avoid duplicates for me.

**Comment by bwssytems** [comment]
Funny thing is I'm having difficulty and it is amazon's fault

https://www.amazonforum.com/s/question/0D54P00006zSyeDSAS/alexa-discovering-smart-devices-from-cache

**Comment by bwssytems** [comment]
What I just found is the alexa app will reconcile devices that are shown multiple times by clicking on the device and waiting for it to figure it out.
